rector-migrate.php

```diff
@@ -0,0 +1,165 @@
+        'dateTimeThisCentury',
+        'dateTimeThisDecade',
+        'dateTimeThisMonth',
+        'dateTimeThisYear',
+        'dayOfMonth',
+        'dayOfWeek',
+        'domainName',
+        'domainWord',
+        'e164PhoneNumber',
+        'email',
+        'emoji',
+        'file',
+        'firefox',
+        'firstName',
+        'firstNameFemale',
+        'firstNameMale',
+        'freeEmail',
+        'freeEmailDomain',
+        'getDefaultTimezone',
+        'hexColor',
+        'hslColor',
+        'hslColorAsArray',
+        'iban',
+        'image',
+        'imageUrl',
+        'imei',
+        'internetExplorer',
+        'iosMobileToken',
+        'ipv4',
+        'ipv6',
+        'iso8601',
+        'jobTitle',
+        'languageCode',
+        'lastName',
+        'latitude',
+        'lexify',
+        'linuxPlatformToken',
+        'linuxProcessor',
+        'localCoordinates',
+        'localIpv4',
+        'locale',
+        'longitude',
+        'macAddress',
+        'macPlatformToken',
+        'macProcessor',
+        'md5',
+        'month',
+        'monthName',
+        'msedge',
+        'name',
+        'numerify',
+        'opera',
+        'paragraph',
+        'paragraphs',
+        'passthrough',
+        'password',
+        'phoneNumber',
+        'postcode',
+        'randomAscii',
+        'randomDigitNotNull',
+        'randomElement',
+        'randomElements',
+        'randomHtml',
+        'randomKey',
+        'randomLetter',
+        'realText',
+        'realTextBetween',
+        'regexify',
+        'rgbColor',
+        'rgbColorAsArray',
+        'rgbCssColor',
+        'rgbaCssColor',
+        'safari',
+        'safeColorName',
+        'safeEmail',
+        'safeEmailDomain',
+        'safeHexColor',
+        'sentence',
+        'sentences',
+        'setDefaultTimezone',
+        'sha1',
+        'sha256',
+        'shuffle',
+        'shuffleArray',
+        'shuffleString',
+        'slug',
+        'streetAddress',
+        'streetName',
+        'streetSuffix',
+        'swiftBicNumber',
+        'text',
+        'time',
+        'timezone',
+        'title',
+        'titleFemale',
+        'titleMale',
+        'tld',
+        'toLower',
+        'toUpper',
+        'unixTime',
+        'url',
+        'userAgent',
+        'userName',
+        'uuid',
+        'windowsPlatformToken',
+        'word',
+        'words',
+        'year',
+    ];
+
+    $methodCalls = [];
+
+    foreach ($properties as $property) {
+        if (!property_exists(\Faker\Generator::class, $property)) {
+            throw new \InvalidArgumentException(sprintf('Property "%s" does not exist in class "%s"', $property, \Faker\Generator::class));
+        }
+
+        $methodCalls[] = new PropertyFetchToMethodCall(Generator::class, $property, $property);
+    }
+
+    $rectorConfig->ruleWithConfiguration(
+        PropertyFetchToMethodCallRector::class,
+        $methodCalls
+    );
```

Comment:
```suggestion
use Faker\Generator;
use PHPStan\PhpDocParser;
use Rector\Config\RectorConfig;
use Rector\Transform\Rector\Assign\PropertyFetchToMethodCallRector;
use Rector\Transform\ValueObject\PropertyFetchToMethodCall;

// This file configures rector/rector to replace all deprecated property usages with their equivalent functions.
return static function (RectorConfig $rectorConfig): void {
    $reflection = new \ReflectionClass(Generator::class);

    $docBlock = $reflection->getDocComment();

    $lexer = new PhpDocParser\Lexer\Lexer();

    $constExprParser = new PhpDocParser\Parser\ConstExprParser();

    $phpDocParser = new PhpDocParser\Parser\PhpDocParser(
        new PhpDocParser\Parser\TypeParser($constExprParser),
        $constExprParser
    );

    $phpDocNode = $phpDocParser->parse(new PhpDocParser\Parser\TokenIterator($lexer->tokenize($docBlock)));

    $properties = array_map(static function (PhpDocParser\Ast\PhpDoc\PropertyTagValueNode $propertyTagValueNode): string {
        if (!preg_match('/^\$(?P<name>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)$/', $propertyTagValueNode->propertyName, $matches)) {
            throw new \RuntimeException(\sprintf(
                'Property name "%s" is not valid.',
                $propertyTagValueNode->propertyName
            ));
        }

        return $matches['name'];
    }, $phpDocNode->getPropertyTagValues());

    $rectorConfig->ruleWithConfiguration(
        PropertyFetchToMethodCallRector::class,
        array_map(static function (string $property): array {
            return new PropertyFetchToMethodCall(Generator::class, $property, $property);
        }, $properties)
    );
```

Perhaps this makes sense after running

```php
composer require phpstan/phpdoc-parser:^1.14.0 --working-dir=vendor-bin/rector
```

?

Reply:
In this case you can only use this config if you have PHPStan running... So I am more for the static version to be honest.

Anyway, all properties are already deprecated and I would not overengineer it.